src/TTextEdit.cpp

```diff
@@ -1421,6 +1426,83 @@ void TTextEdit::slot_copySelectionToClipboardHTML()
     return;
 }
 
+void TTextEdit::slot_copySelectionToClipboardImage()
+{
+    //this->grab().save(QStringLiteral("/tmp/image.png"));
+
+    //    const QRect& rect = e->rect();
+
+    // mPA QPoint where selection started
+    // mPB QPoint where selection ended
+
+    // if selection was made backwards swap
+    // right to left
+    if ((mPA.y() == mPB.y()) && (mPA.x() > mPB.x())) {
+        swap(mPA, mPB);
+    }
+    // down to up
+    if (mPA.y() > mPB.y()) {
+        swap(mPA, mPB);
+    }
+
+    if (mFontWidth <= 0 || mFontHeight <= 0) {
+        return;
+    }
+
+    if (mScreenHeight <= 0 || mScreenWidth <= 0) {
+        mScreenHeight = height() / mFontHeight;
+        mScreenWidth = 100;
+        if (mScreenHeight <= 0 || mScreenWidth <= 0) {
+            return;
+        }
+    }
+
+    auto height = (mPB.y() - mPA.y()+1)*mFontHeight;
+
+    // find the biggest width of text we need to work with
+    int characterWidth = 0;
+    for (int y = mPA.y(), total = mPB.y() + 1; y <= total; ++y) {
+        auto lineWidth = static_cast<int>(mpBuffer->buffer[y].size());
+        characterWidth = lineWidth > characterWidth ? lineWidth : characterWidth;
+    }
+
+
+    auto width = characterWidth*mFontWidth;
```

Comment:
How about:
```
auto width = qMin(1'000'000, characterWidth*mFontWidth);
```
then you do not need the separate limit test afterwards? :grinning: